sim-lib/src/lib.rs

```diff
@@ -549,7 +549,16 @@ impl Simulation {
             listener.clone(),
         ));
 
+        let result_logger = Arc::new(Mutex::new(PaymentResultLogger::new()));
+
+        tasks.spawn(run_results_logger(
+            listener.clone(),
+            result_logger.clone(),
+            Duration::from_secs(60),
```

Comment:
We may want to make this configurable and default it to 1 min

Reply:
Clippy starts getting angry if we have too many args for `Simulation::new` - pushed example in https://github.com/bitcoin-dev-project/sim-ln/pull/141/commits/e929a2d6c28687df4b491594dc84c0ae9a1dc208 which adds a `SimulationConfig` struct to house all of our various flags. 

If we like it, I'll break that up into a refactor commit + add some validation on the config so that we fail if we're given bad config values (eg, zero expected amount).